Content.Server/Body/Systems/RespiratorSystem.cs

```diff
@@ -59,6 +61,13 @@ private void OnUnpaused(Entity<RespiratorComponent> ent, ref EntityUnpausedEvent
         ent.Comp.NextUpdate += args.PausedTime;
     }
 
+    public bool CanBreathe(EntityUid uid)
+    {
+        if (TryComp<PullableComponent>(uid, out var pullable) && pullable.GrabStage == GrabStage.Suffocate)
+            return false;
+        return true;
+    }
+
```

Comment:
CanBreathe if anything should be an event and grab should not be infecting other systems.